compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -444,6 +459,32 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
     return true;
   }
 
+  bool visit(luci::CircleConcatenation *node)
+  {
+    const auto num_values = node->numValues();
```

Comment:
I think we need to check rank of node and if not 4 then return falsel;

Reply:
node rank is checked before the visitor is called.

Related code
```cpp
...
     ConvertNCHWToNHWC converter;
      auto circle_node = loco::must_cast<luci::CircleNode *>(node);
      if (circle_node->rank() != 4)
        continue;

      if (circle_node->accept(&converter))
      {
        set_data_format(node, DataFormat::NHWC);
        changed = true;
      }
...
```